app/views/govuk_publishing_components/components/_search.html.erb

```diff
@@ -32,11 +28,31 @@
     classes << "gem-c-search--on-white"
   end
   classes << "gem-c-search--separate-label" if local_assigns.include?(:inline_label) or local_assigns.include?(:label_size)
```

Comment:
A teeny tiny nitpick: you've mentioned why it's a bad idea to specify the `inline_label` using the `||=` convention at the top (because it's falsey, ruby will defer to the default value). Do you think its worth adding a comment to this effect? I'm thinking of a well-meaning dev seeing that it's not included at the top and adding it in, potentially breaking a few things.

I'm asking instead of suggesting partly because the tests would catch this.

Reply:
Ooh, this is a can of worms. I agree, it's a thing that might be overridden without someone meaning to - so we could suggest moving away from using `false ` values as the default.

For this component, I'd also suggest that renaming the `inline_label` to `label_position` of which the options are `inline` or `above` (or something similar, just throwing ideas out here).  Then we can set the default to be `inline` using the `||=`, run the two parameters in parallel while we update the component usage, and then deprecate it. I think there are a couple of breaking changes coming up, which is good timing.

I'll raise an issue to gently nudge a discussion around this and see what the community consensus is.

Reply:
Issue raised: https://github.com/alphagov/govuk_publishing_components/issues/2472